### PR TITLE
chore: named-exports in core package

### PR DESCRIPTION
**Pull request by juliusmarminge**
just converts `export * from 'x'` to export everything but named from all public entrypoints (they which are specified in rollup config) (in core package only)

### CODE REVIEW

**Comment by juliusmarminge** [comment]
Current dependencies on/for this PR:
* `next`
  * **PR #5273** <a href="https://app.graphite.dev/github/pr/trpc/trpc/5273?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/trpc/trpc/5273?utm_source=stack-comment).

**Review by KATT — APPROVED** [review]

File: packages/core/src/observable/index.ts [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+export type { inferObservableValue } from './observable';
+export type {
+  MonoTypeOperatorFunction,
+  Observable,
+  Observer,
+  OperatorFunction,
+  Subscribable,
+  SubscriptionLike,
+  TeardownLogic,
+  UnaryFunction,
```
Comment (KATT) [inline packages/core/src/observable/index.ts:11]: this feels and `MonoTypeOperatorFunction` like an internal thing

unsure if they're needed to be exported at all

Comment (KATT) [reply packages/core/src/observable/index.ts:11]: fun note - I don't know what Unary or MonoType really means, i just copied naming from RxJs :D

Comment (KATT) [reply packages/core/src/observable/index.ts:11]: i know computers

bailey-dog-meme-2

**Pull request merged**